File: atcoder/abc178_f.cpp
// https://atcoder.jp/contests/abc176/tasks/abc178_f
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
// #include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define drep(i,n) for(int i = (n-1); i >= 0; i--)
#define all(v) (v).begin(),(v).end()
#define maxs(x,y) (x = max(x,y))
#define mins(x,y) (x = min(x,y))
template <class T> bool chmax(T &a, const T &b) { if (a < b) { a = b; return 1; } return 0; }
template <class T> bool chmin(T &a, const T &b) { if (b < a) { a = b; return 1; } return 0; }
template <class T> T gcd(T a, T b) { return b ? gcd(b, a % b) : a; }
template <class T> T lcm(T a, T b) { return a/gcd(a,b)*b; }
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;

int main() {
    cin.tie(nullptr);
    ios::sync_with_stdio(false);
	int n;
    cin >> n;
    vector<int> a(n), b(n), cnta(n), cntb(n), cuma(n+1), cumb(n+1);
    rep(i,n) {
        cin >> a[i];
        cnta[a[i]-1]++;
    }
    rep(i,n) {
        cin >> b[i];
        cntb[b[i]-1]++;
    }
    rep(i,n) if(cnta[i]+cntb[i]>n) {
        cout << "No" << endl;
        return 0;
    }
    rep(i,n) {
        cuma[i+1] = cuma[i] + cnta[i];
        cumb[i+1] = cumb[i] + cntb[i];
    }

    int x = 0;
    rep(i,n) {
        x = max(x, cuma[i+1] - cumb[i]);
    }
    // cout << x << endl;
    cout << "Yes" << endl;
    rep(i,n) cout << b[(i+n-x)%n] << ' ';
    cout << endl;
	return 0;
}


// WA
// int main() {
//     cin.tie(nullptr);
//     ios::sync_with_stdio(false);
// 	int n;
//     cin >> n;
//     vector<int> a(n), b(n), cnta(n), cntb(n), posa(n), posb(n);
//     rep(i,n) {
//         cin >> a[i];
//         --a[i];
//         cnta[a[i]]++;
//         posa[a[i]] = i;
//     }
//     rep(i,n) {
//         cin >> b[i];
//         --b[i];
//         cntb[b[i]]++;
//         posb[b[i]] = i;
//     }
//     int maxi, maxcnt = 0;
//     rep(i,n) {
//         if(cnta[i]+cntb[i]>maxcnt) {
//             maxcnt = cnta[i]+cntb[i];
//             maxi = i;
//         }
//         if(cnta[i]+cntb[i]>n) {
//             cout << "No" << endl;
//             return 0;
//         }
//     }
//     cout << "Yes" << endl;
//     int slide = posa[maxi] - posb[maxi] + cntb[maxi];
//     rep(i,n) {
//         cout << b[(i+slide)%n]+1 << ' ';
//     }

//     cout << endl;
// 	return 0;
// }

File: atcoder/abc178_f.py
# https://atcoder.jp/contests/abc177/tasks/abc178_e

def main():
    n = int(input())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    c, d = [0]*(n+1), [0]*(n+1)
    for i in range(n):
        c[a[i]] += 1
        d[b[i]] += 1
    for i in range(1,n+1):
        if c[i]+d[i]>n:
            print("No")
            return
    for i in range(1,n+1):
        c[i] = c[i] + c[i-1]
        d[i] = d[i] + d[i-1]
    x = 0
    for i in range(1,n+1):
        x = max(x, c[i]-d[i-1])
    print("Yes")
    ans = b[n-x:] + b[:n-x]
    print(*ans)
    return

main()



# WA
# def main():
#     from collections import Counter
#     import bisect
#     n = int(input())
#     a = list(map(int, input().split()))
#     b = list(map(int, input().split()))
#     cnta = Counter(a)
#     cntb = Counter(b)
#     for k,v in cnta.items():
#         if v + cntb[k] > n:
#             print("No")
#             return
#     print("Yes")
#     ans = [0]*n
#     for kb, vb in sorted(cntb.items(), key=lambda x:x[1], reverse=True):
#         for ka, va in sorted(cnta.items(), key=lambda x:x[1], reverse=True):
#             if ka==kb: continue
#             r = bisect.bisect_right(a, ka)
#             chg = min(va, vb)
#             for i in range(chg):
#                 ans[r-va+i] = kb
#             vb -= chg
#             cnta[ka] -= chg
#             if vb==0: break
#     print(*ans)
#     return

# main()
